### PR TITLE
feat: Adds UDF regexp_replace

### LINKED ISSUE #5491: Add regexp_replace function
`regexp_replace(s, regex, replace)` - Given an input string `s`, replace all matches of `regex` with the string `replace`.

### PR DESCRIPTION
**Pull request by AlanConfluent**
### Description 
Adds the a new UDF regexp_replace that replaces all matches of a regexp in a string with a new substring.

Fixes #5491 

### Testing done 
Wrote unit tests.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/ksqldb-reference/scalar-functions.md [hunk 1]
```diff
@@ -415,6 +415,14 @@ the entire substring is returned by default.
 For example, `REGEXP_EXTRACT("(.*) (.*)", 'hello there', 2)`
 returns "there".
 
+### `REGEXP_REPLACE`
+
+```sql
+REGEXP_REPLACE(col1, 'a.b+', 'bar')
+```
+
+Replace all matches of a regex in a string with a new string.
```
Comment (big-andy-coates) [inline docs/developer-guide/ksqldb-reference/scalar-functions.md:424]: Document null handling

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/replace.json [hunk 2]
```diff
@@ -23,6 +23,23 @@
...
+      "inputs": [
+        {"topic": "test_topic", "value": {"input_string": "baababaa"}},
+        {"topic": "test_topic", "value": {"input_string": "baabbabaa"}},
+        {"topic": "test_topic", "value": {"input_string": null}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "value": {"EXTRACTED": "bacataa"}},
+        {"topic": "OUTPUT", "value": {"EXTRACTED": "bacatbcataa"}},
+        {"topic": "OUTPUT", "value": {"EXTRACTED": null}}
+      ]
```
Comment (big-andy-coates) [inline ksqldb-functional-tests/src/test/resources/query-validation-tests/replace.json:42]: Would be good if this test case covered nulls being passed to all params.

File: docs/developer-guide/ksqldb-reference/scalar-functions.md (hunk 1 above)
Comment (AlanConfluent) [reply docs/developer-guide/ksqldb-reference/scalar-functions.md]: Done

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/replace.json (hunk 2 above)
Comment (AlanConfluent) [reply ksqldb-functional-tests/src/test/resources/query-validation-tests/replace.json]: Done

**Pull request merged**